src/phoenix/experimental/callbacks/langchain_tracer.py

```diff
@@ -0,0 +1,37 @@
+from typing import List
+from uuid import uuid4
+
+from langchain.callbacks.tracers.base import BaseTracer
+from langchain.callbacks.tracers.schemas import Run
+
+from phoenix.trace.schemas import Span, SpanContext, SpanKind, SpanStatusCode
+
+
+class OpenInferenceTracer(BaseTracer):  # type: ignore
+    def __init__(self) -> None:
+        self.spans: List[Span] = []
+        self.trace_id = uuid4()
+        super().__init__()
+
+    def _persist_run(self, run: Run) -> None:
+        if run.parent_run_id is None:
+            self.trace_id = uuid4()
+        span = Span(
+            name=run.name,
+            span_kind=SpanKind(run.run_type.upper()),
```

Comment:
this assumes a 1-1 isomorphism between run_type and spanKind. We need to provide some sort of compatibility layer so we don't tie ourselves to langchain too heavily here. Thoughts on how these map?

Reply:
currently they're mapped 1-1 with langchain. but langchain is moving away from enums and to arbitrary strings, so we may have to do the same